### PR TITLE
Moved the Publish method from runbookDraft to Runbook.

### PR DESCRIPTION
**Pull request by khughes66**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process.

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x] My spec meets the review criteria:
  - [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] The spec follows the guidelines described in the [Swagger checklist](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md) document.
  - [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Comment by khughes66** [comment]
Martin, can you please review? Thanks.

**Comment by khughes66** [comment]
@marstr Please review

File: specification/automation/resource-manager/Microsoft.Automation/stable/2018-01-15/dscNodeCounts.json [hunk 1]
```diff
@@ -140,7 +140,8 @@
       "x-ms-enum": {
         "name": "CountType",
         "modelAsString": true
-      }
+      },
+      "x-ms-parameter-location": "method"
```
[the remark this answers is not in the document]
Comment (vrdmr) [reply specification/automation/resource-manager/Microsoft.Automation/stable/2018-01-15/dscNodeCounts.json]: Hi @marstr  - The SDK hasn't been released and we are correcting this parameter as this should be a method level parameter for the `NodeCountInformation` call.

**Comment by vrdmr** [comment]
@ravbhatnagar Could you please take a look at the PR.

**Review by marstr — CHANGES_REQUESTED** [review]
Looks like both some linter and model errors have popped up, and there are now conflicts with the master branch. Please take a look here:
https://travis-ci.org/Azure/azure-rest-api-specs/jobs/401942873#L775

**Comment by vrdmr** [comment]
@khughes66 - Can you check `JobStreamProperties` in the runbook, and make sure they are same - [Travis Model issues](https://travis-ci.org/Azure/azure-rest-api-specs/jobs/401942873#L4822-L4928).

**Comment by ravbhatnagar** [comment]
Looks good. The change is to expose publish action on runbooks instead of draft. This change was discussed in person with ARM.

**Comment by marstr** [comment]
I'm going to be on vacation until Monday, July 30th. If this PR needs attention in the interim, please contact my boss, @salameer.

**Comment by vrdmr** [comment]
@marstr  When running the autorest --python locally, it does not throw the `duplicate definitions for JobStreamProperties` error. Could you please take a look at this issue?

```
varads-mbp:resource-manager varadmeru$ autorest readme.md --python --output-folder="."
AutoRest code generation utility [version: 2.0.4283; node: v10.3.0]
(C) 2018 Microsoft Corporation.
https://aka.ms/autorest
   Loading AutoRest core      '/Users/varadmeru/.autorest/@microsoft.azure_autorest-core@2.0.4283/node_modules/@microsoft.azure/autorest-core/dist' (2.0.4283)
   Loading AutoRest extension '@microsoft.azure/autorest.python' (~2.1.26->2.1.40)
   Loading AutoRest extension '@microsoft.azure/autorest.modeler' (2.3.44->2.3.44)
varads-mbp:resource-manager varadmeru$ 
```

**Comment by marstr** [comment]
Looking at the build logs here: https://travis-ci.org/Azure/azure-rest-api-specs/jobs/413755232#L613 it says that there are duplicated definitions of JobStreamProperties, which conflict. Given that they're in separate API Versions, **2018-01-15** and **2017-05-15-preview**.

In the command you executed above, you didn't provide the flag `--multiapi`, which means AutoRest will only process the one default tag in the readme. I suspect there's a tag in README.md which accidentally includes both `specification/automation/resource-manager/Microsoft.Automation/stable/2018-01-15/dscCompilationJob.json` and `specification/automation/resource-manager/Microsoft.Automation/preview/2017-05-15-preview/job.json`.

Let me know what you find!

**Comment by vrdmr** [comment]
@khughes66 Could you please close this PR. #3646 fixes the issues and passes.

**Comment by khughes66** [comment]
There were some strange issues with this PR (see the checks above).  We duplicated it into PR  https://github.com/Azure/azure-rest-api-specs/pull/3646, which has passed the checks.  Closing this PR in favor of that one.

**Pull request closed without merging**